trunk/SUAVE/Components/Energy/Networks/Propulsor_Surrogate.py

```diff
@@ -0,0 +1,138 @@
+        self.input_file        = None
+        self.sfc_surrogate     = None
+        self.thrust_surrogate  = None
+        self.areas             = Data()
+    
+    # manage process with a driver function
+    def evaluate_thrust(self,state):
+        
+        # Unpack the surrogate
+        sfc_surrogate = self.sfc_surrogate
+        thr_surrogate = self.thrust_surrogate
+        
+        # Unpack the conditions
+        conditions = state.conditions
+        altitude   = conditions.freestream.altitude
+        mach       = conditions.freestream.mach_number
+        throttle   = conditions.propulsion.throttle
+        
+        cond = np.hstack([altitude,mach,throttle])
+        
+        # Run the surrogate for a range of altitudes
+        data_len = len(altitude)
+        sfc = np.zeros([data_len,1])  
+        thr = np.zeros([data_len,1]) 
+        for ii,_ in enumerate(altitude):
+            sfc[ii] = sfc_surrogate.predict(np.array([altitude[ii][0],mach[ii][0],throttle[ii][0]]))    
+            thr[ii] = thr_surrogate.predict(np.array([altitude[ii][0],mach[ii][0],throttle[ii][0]]))   
+            
+        ## Test plot, run a range of throttles
+        #throttles = np.linspace(0.,1.)
+        #alts  = 10000.*np.ones_like(throttles)
+        #machs = 0.9*np.ones_like(throttles)
+        
+        #thrusts = thr_surrogate.predict(np.array([alts,machs,throttles]).T)   
+        
+        #import pylab as plt
+        
+        #plt.figure("throttles")
+        #axes = plt.gca()  
+        #axes.plot(throttles,thrusts)
+        
+        #plt.show()
+        
+        
+        F    = thr
+        mdot = thr*sfc*self.number_of_engines
+        
+        # Save the output
+        results = Data()
+        results.thrust_force_vector = self.number_of_engines * F * [1,0,0] 
+        results.vehicle_mass_rate   = mdot
+    
+        return results          
+    
+    def build_surrogate(self):
+        
+        # file name to look for
+        file_name = self.input_file
+        
+        # Load the CSV file
+        my_data = np.genfromtxt(file_name, delimiter=',')
+        
+        # Clean up to remove redundant lines
+        # Section 1
+        #col1 = my_data[1:,:1]
+        #col2 = my_data[1:,1:2]
+        #col3 = my_data[1:,2:3]
+        #for ii in xrange(0, len(col1)-1):
+            #if (col1[ii] == col1[ii+1])&(col2[ii] == col2[ii+1])&(col3[ii] == col3[ii+1]):
+                #np.delete(my_data,(ii), axis=0)
+
+        b = np.ascontiguousarray(my_data).view(np.dtype((np.void, my_data.dtype.itemsize * my_data.shape[1])))
+        _, idx = np.unique(b, return_index=True)
+        
+        my_data = my_data[idx]                
+                
+    
+        xy  = my_data[1:,:3] # Altitude, Mach, Throttle
+        thr = np.transpose(np.atleast_2d(my_data[1:,3])) # Thrust
+        sfc = np.transpose(np.atleast_2d(my_data[1:,4]))  # SFC
+        
+
+        # Pick the type of process
```

Comment:
Consider making this an optional input argument rather than a set of commented out options.